Fix: 틈 요청 첫 수락 시 요청자 스케줄 생성

yangjiae12 opened the pull request:
### 👀 관련 이슈

<!-- 관련 이슈를 적어주세요 -->
#94 

### ✨ 작업한 내용

<!-- 작업한 내용을 적어주세요 -->
- 기존에 틈 요청 첫 수락 시에 수신자에게만 스케줄이 생기던 문제 해결
   - 수신자가 틈 요청을 처음 수락하는 경우 요청자와 수신자 모두에게 스케줄 생성
   - 이후 다른 수신자들이 수락할 경우 수신자 본인에게만 스케줄 생성

### 🌀 PR Point

<!-- 코드리뷰가 필요한 부분이 있다면 적어주세요 -->

### 🍰 참고사항

<!-- 참고할 사항이 있다면 적어주세요 -->
- 동시성 문제가 발생할 수 있을 것 같은데, 이 부분은 이전에 약속된 틈 취소의 동시성 문제와 같이 별도의 이슈에서 해결하도록 하겠습니다.
- 중복 처리에 관련한 기획에 따라 틈 응답 시에도 검증 로직이 필요한데 이것도 별도의 이슈에서 진행하도록 하겠습니다.

### 📷 스크린샷 또는 GIF

| 기능 | 스크린샷 |
| :--: | :------: |
| 첫 수락 (1번이 응답 -> 요청자인 2번의 스케줄도 생성) |<img width="1804" height="129" alt="image" src="https://github.com/user-attachments/assets/6ddacc91-204e-4228-af30-e263b4eef8e1" />|
| 첫 수락 X (1번이 응답 -> 1번의 스케줄만 생성) |<img width="1817" height="194" alt="image" src="https://github.com/user-attachments/assets/76920028-4f5d-4a3b-b064-172a6b0fa28d" />|

yangjiae12 committed:
Fix: 수신자 첫 수락 시 요청자와 수신자 모두에게 스케줄 생성

- 수신자가 틈 요청을 처음 수락하는 경우 요청자와 수신자 모두에게 스케줄 생성
- 이후 다른 수신자들이 수락할 경우 수신자 본인에게만 스케줄 생성

yangjiae12 committed:
Fix: 틈 스케줄 생성 시 includeTeum = true로 명시

- 기본값이 false인 includeTeum을 명시적으로 true로 설정
- 기타 기본값 필드는 생략하여 builder 코드 간결화

yangjiae12 committed:
Merge branch 'develop' into fix/94-teum-schedule-create

yangjiae12 commented:
틈 스케줄에서는 빈틈 시간 포함이 기본이라고 하여 수정하였습니다! 더 수정할 것 없으니 리뷰 해주시면 감사하겠습니다~~

Jinseong01 commented on src/main/java/umc/teumteum/server/domain/teum/service/TeumServiceImpl.java:
```diff
@@ -152,18 +152,30 @@ public TeumStatusUpdateResponseDto updateResponseStatus(Long responseId, Long us
+            scheduleRepository.save(receiverSchedule);
+
+            // 스케줄이 처음 생성되는 경우에만 요청자도 생성
+            if (!hasExistingSchedules) {
```
흐름상 크게 문제될 부분은 없는거 같습니다.
동시성..은 추후에 시간남을때 천천히 진행해도 될 거 같습니다.. (저도 안해본 입장이라..)

다만, `hasExistingSchedules`값이 `!request.getSchedules().isEmpty();`로 초기화되고 if문에서 다시 `NOT 연산`을 하는데 이 부분만 수정해주시면 될 거 같습니다!

yangjiae12 replied:
수정 완료했습니다! 피드백 감사합니다~

yangjiae12 committed:
Refactor: 불필요한 부정 조건 제거하고 스케줄 여부 직접 체크

- hasExistingSchedules 변수 제거
- if (request.getSchedules().isEmpty()) 조건으로 간결하게 처리

Jinseong01 reviewed (approved):
수고하셨습니다!!
